reinforcement_learning/rlclientlib/logger/event_batcher.h

```diff
@@ -0,0 +1,87 @@
+#include "event_queue.h"
+#include "ranking_event.h"
+#include "generated/RankingEvent_generated.h"
+#include "generated/OutcomeEvent_generated.h"
+
+namespace reinforcement_learning {
+  namespace utility {
+    template<typename TEvent>
+    class event_batcher {
+    public:
+      event_batcher() {};
+      void batch_serialize(data_buffer& oss, size_t& remaining, event_queue<TEvent>& queue, size_t _send_high_water_mark, size_t batch_size);
+    };
+
+    template<typename TEvent>
+    void event_batcher<TEvent>::batch_serialize(data_buffer& oss, size_t& remaining, event_queue<TEvent>& queue, size_t _send_high_water_mark, size_t batch_size) {
+
+      if (batch_size <= 0) {
+        throw("event_batcher::batch_serialize batch_size is 0 - message is too large.");
+      }
+
+      if (std::is_same<TEvent, ranking_event>::value) {
+        std::vector<flatbuffers::Offset<RankingEvent>> events_offset;
+        flatbuffers::FlatBufferBuilder builder(_send_high_water_mark);
```

Comment:
also, can you make flatbuffers not visible outside of ranking_event / interaction event level?